### PR TITLE
bugfix: checked `consumer` first, it maybe a boolean value when it was beed deleted.

### PR DESCRIPTION
**Pull request by membphis**

### CODE REVIEW

**Comment by membphis** [comment]
fix https://github.com/apache/incubator-apisix/issues/1138

File: lua/apisix/consumer.lua [hunk 1]
```diff
@@ -35,6 +36,10 @@ local function plugin_consumer()
     end
 
     for _, consumer in ipairs(consumers.values) do
+        if type(consumer) ~= "table" then
```
Comment (moonming) [inline lua/apisix/consumer.lua]:
```
if type(consumer) == "table" then
    for ...
end
```
is better.

**Comment by moonming** [comment]
Will other objects have similar problem? for example delete route

**Comment by membphis** [comment]
> Will other objects have similar problem? for example delete route

https://github.com/apache/incubator-apisix/blob/master/lua/apisix/core/config_etcd.lua#L284

the old object  will be set to boolean value if it was deleted or expired.

File: lua/apisix/consumer.lua (hunk 1 above)
Comment (membphis) [reply lua/apisix/consumer.lua]: The current code style is to reduce indentation. I prefer the current coding style.

**Pull request merged**